src/components/withScrollFader/index.tsx

```diff
@@ -0,0 +1,173 @@
+import React, {useCallback} from 'react';
+import {
+  StyleSheet,
+  // eslint-disable-next-line no-unused-vars
+  FlatListProps,
+  // eslint-disable-next-line no-unused-vars
+  ScrollViewProps,
+  // eslint-disable-next-line no-unused-vars
+  NativeSyntheticEvent,
+  // eslint-disable-next-line no-unused-vars
+  NativeScrollEvent
+} from 'react-native';
+// eslint-disable-next-line no-unused-vars
+import forwardRef, {ForwardRefInjectedProps} from '../../commons/forwardRef';
+import withScrollReached, {
+  // eslint-disable-next-line no-unused-vars
+  WithScrollReachedProps,
+  // eslint-disable-next-line no-unused-vars
+  WithScrollReachedOptionsProps
+} from '../../commons/withScrollReached';
+import View from '../view';
+import Image from '../image';
+import Assets from '../../assets';
+
+export type ScrollFaderProps = {
+  onScroll: (event: NativeSyntheticEvent<NativeScrollEvent>) => void;
+  /**
+   * Use to render the fade
+   */
+  renderFader: () => React.ReactNode;
```

Comment:
If the whole idea of this HOC is to provide a render function that renders a component based on props. wouldn't it be easier to create a component? 
I think it would be a lot simpler if the use can just render a "ScrollFader" component. WDYT?

Reply:
Maybe if there has to be a **single** child that has to have `onScroll` (or has to be a `supportedView`).

Reply:
Changed to Fader, changed the title (but not the branch name :) )